src/main/java/org/commonjava/util/sidecar/jaxrs/PreSeedResource.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (C) 2011-2021 Red Hat, Inc. (https://github.com/Commonjava/indy-sidecar)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.util.sidecar.jaxrs;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.http.HttpServerRequest;
+import org.apache.commons.io.FileUtils;
+import org.commonjava.util.sidecar.services.ArchiveRetrieveService;
+import org.commonjava.util.sidecar.services.ProxyService;
+import org.commonjava.util.sidecar.util.TransferStreamingOutput;
+import org.commonjava.util.sidecar.util.UrlUtils;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.HEAD;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.InputStream;
+import java.util.Optional;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_OCTET_STREAM;
+import static org.commonjava.util.sidecar.services.ArchiveRetrieveService.isDecompressed;
+import static org.commonjava.util.sidecar.services.ProxyService.doProxy;
+import static org.eclipse.microprofile.openapi.annotations.enums.ParameterIn.PATH;
+
+@Path( "/api/folo/track/{id}/maven/{type: (hosted|group|remote)}/{name}" )
+public class PreSeedResource
+{
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    ProxyService proxyService;
+
+    @Inject
+    ArchiveRetrieveService archiveService;
+
+    @HEAD
+    public Uni<Response> head( @PathParam( "path" ) String path, final @Context HttpServerRequest request )
```

Comment:
Seems "path" is missing here?

Reply:
@ligangty , this is migrated from original ProxyResource(Renaming to PreSeed* is more reasonable here), I don't handle the head stuff as the current build functions I pick is heading to GET function.